### DISCUSSION TITLE
Arranging activities dynamically based on user request

### DISCUSSION THREAD

**Discussion by krishnaraman001** [uber/cadence#4651] [Q&A]
I've a requirement where I need to arrange my activities dynamically
Let's say I've three activities:
Capitalize Text
Replace Text
Split Text
Now, I **need to arrange these activities dynamically based on user request** and also pass result of an activity to its consecutive activity
So, I'm developing UI where user can drag and drop these activities and connect them as he wants. Planning to use Cadence as backend for this
How can I achieve this with Cadence in java?

**Comment by longquanzheng** [comment]
> on user request and also pass result of an activity to its consecutive activity

Depends on how you are passing the user input and  you can use `newUntypedActivityStub` to execute activity based on the input. 

For example, if the user input is passed in as a signal, then it can contain the activity type and the activity input, then the signal can invoke the activity using the untyped activity stub.

https://www.javadoc.io/static/com.uber.cadence/cadence-client/3.5.1/com/uber/cadence/workflow/Workflow.html#newUntypedActivityStub-com.uber.cadence.activity.ActivityOptions-

It should be very straightforward. I will find some time to write a sample code if you really need it.
